api/src/main/java/org/pmiops/workbench/api/CohortBuilderController.java

```diff
@@ -47,11 +47,12 @@
             final Criteria criteria = new Criteria();
             criteria.setId(row.get(rm.get("id")).getLongValue());
             criteria.setType(row.get(rm.get("type")).getStringValue());
-            criteria.setCode(row.get(rm.get("code")).getStringValue());
+            criteria.setCode(row.get(rm.get("code")).isNull() ? null : row.get(rm.get("code")).getStringValue());
```

Comment:
If `code` is nullable, we'll want to remove it from criteria/required in `workbench.yaml`.  Also, why _is_ it nullable?  I believe you that it is, I'm just curious.  Is this the kind of thing where the criteria has to have at least one of `code` or `conceptId`?

Reply:
Demographics don't have codes. Only icd9, icd10 and cpt have codes, but tables are structured the same for the sake of making the sql consistent across criteria tables.

Reply:
Why not just combine the criteria tables (with an extra column for ICD9 vs. something else)?

Reply:
Well... some of the tables aren't exactly the same but very close. I think it's easier for them the be in separate table for ease of rebuilding the tables and calculating the counts.

Reply:
OK. My guess is that if the table structure is very similar, and the way you query it is very similar, the code will be simpler and easier to maintain if you use a single table (with optional columns that are only populated for certain types of criteria.)

Rebuilding parts of the table shouldn't be that hard to pull off... the tools that would normally populate a whole table will instead delete and insert rows in a subset of it.

Anyway, something to think about when you're working on unified data / API for criteria with the data browser team.

Reply:
(I wouldn't necessarily change anything now if we're planning on changing the representation / switching to Cloud SQL moving forward.)